src/newrelic.coffee

```
@@ -314,12 +374,99 @@ plugin = (robot) ->
+        room = msg.envelope.room
+        subscriber = msg.envelope.room
+        subscriptions = robot.brain.get "newrelicviolations_subscriptions"
+
+        # I'm sure there's a more idiomatic way of doing this:
+        my_subs = [['id', 'field', 'pattern']]
+        for id, details of subscriptions
+            if details.subscriber == subscriber
+                my_subs.push [id, details.field, details.pattern]
+        if my_subs.length > 1
+            message_room robot, room, mdTable my_subs
+        else
+            message_room robot, room, "You are not subscribed to any alerts"
+
+   robot.respond /(newrelic|nr) alerts* unsubscribe (.*)$/i, (msg) ->
+        subscription_id  = msg.match[2]
+        room = msg.envelope.room
+
+        subscriptions = robot.brain.get "newrelicviolations_subscriptions"
+
+        subscription = subscriptions[subscription_id]
+
+        if not subscription
+            message_room robot, room, "No such subscription!"
+            return
+
+        if subscription.subscriber == room
+            delete subscriptions[subscription_id]
+            message_room robot, room, "subscription #{subscription_id} deleted!"
+        else
+            message_room robot, room, "It doesn't seem like that subscription belongs to you"
+
+
+
+   robot.respond /(newrelic|nr) alerts* subscribe ([\w\*\.\_\-]+)\ *([\w\*\.\_\-]*)$/i, (msg) ->
+        if msg.match[3] != ''
+          field = msg.match[2]
+          pattern = msg.match[3]
+        else
+          # if there's only one match, then the first argument is
+          # the pattern, and we assume the field is policy_name
+          field = 'policy_name'
+          pattern = msg.match[2]
+
+        subscriber = msg.envelope.room
+        subscription_id = uniqueId()
+
+        subscriptions = robot.brain.get "newrelicviolations_subscriptions"
+
+        if not subscriptions
+            subscriptions = {}
+
+        subscription = {pattern,field,subscriber}
```

I did not know this syntax, very nice

---

Coffeescript might be growing on me, except for where my unintentional pythonisms cause weird or silent errors (like starting a block with a colon, or 'for x in y' instead of 'for x of y')

---

yep, for all the complaining I might do about CS syntax, it does have some nice features